frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/homepage/MyProjectsPortlet.java

```diff
@@ -121,13 +126,18 @@ private void serveProjectList(ResourceRequest request, ResourceResponse response
         PaginationParameters paginationParameters = PaginationParser.parametersFrom(originalServletRequest);
         PortletUtils.handlePaginationSortOrder(request, paginationParameters, projectFilteredFields, PROJECT_NO_SORT);
 
-        String rolesSelected = request.getParameter("roles");
-        List<Boolean> listOfRolesSelected = Arrays.stream(rolesSelected.split(","))
+        String rolesAndClearingStateSelected = request.getParameter("rolesandclearingstate");
+        List<Boolean> listOfRolesAndClearingStateSelected = Arrays.stream(rolesAndClearingStateSelected.split(","))
                 .map(role -> Boolean.parseBoolean(role)).collect(Collectors.toList());
         Boolean userChoice = Boolean.parseBoolean(request.getParameter("userChoice"));
         Map<String, Boolean> userRoles = new HashMap<>();
         for (int i = 0; i < listOfRoles.size(); i++) {
-            userRoles.put(listOfRoles.get(i), listOfRolesSelected.get(i));
+            userRoles.put(listOfRoles.get(i), listOfRolesAndClearingStateSelected.get(i));
+        }
+        
+        Map<String, Boolean> clearingState = new HashMap<>();
+        for (int j = 0; j < listOfClearingState.size(); j++) {
+        	clearingState.put(listOfClearingState.get(j), listOfRolesAndClearingStateSelected.get(j+7));
```

Comment:
Please use `listOfRoles.size()` instead of hardcoded value(`7`)

Reply:
Created a new variable for size and removed the hard coded value.